enginetest/join_op_tests.go

```diff
@@ -106,6 +106,102 @@ var joinCostTests = []struct {
 			setup.XyData[0],
 		},
 		tests: []JoinOpTests{
+			{
+				Query: "select * from mytable where exists (select * from mytable where i = 1);",
+				Expected: []sql.Row{
+					{1, "first row"},
+					{2, "second row"},
+					{3, "third row"},
+				},
+			},
+			// queries that test subquery hoisting
+			{
+				// case 1: condition uses columns from both sides
```

Comment:
Thanks for adding in these comments to explain the intent of the test. That is super super helpful for understanding what the tests are covering.

Reply:
@PavelSafronov deserves credit for this and the tests :)